### PR TITLE
Add Symphony CRD

### PR DESCRIPTION
**Pull request by jveski**
Symphony allows sets of compositions to easily be spawned given a list of synthesizers. Kind of like a "meta composition" - a composition of compositions.

### CODE REVIEW

File: internal/controllers/replication/symphony.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	})
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("listing existing compositions: %w", err)
+	}
+
+	modified, err := c.syncStatus(ctx, symph, existing)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if modified {
```
Comment (AYM1607) [inline internal/controllers/replication/symphony.go:54]: What's the reasoning behind using modified as a signal to stop reconciliation of these sub-operations? Is it worth adding comments?

File: internal/controllers/replication/symphony.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+	existingBySynthName, modified, err := c.reconcileReverse(ctx, symph, existing)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if modified {
+		return ctrl.Result{}, nil
+	}
+
+	if symph.DeletionTimestamp == nil {
+		modified, err := c.reconcileForward(ctx, symph, existingBySynthName)
```
Comment (AYM1607) [inline internal/controllers/replication/symphony.go:76]: Comments on why we need to reconcile forwards and backwards?

File: internal/controllers/replication/symphony.go (hunk 1 above)
Comment (jveski) [reply internal/controllers/replication/symphony.go:54]: Passing down a bool is kind of strange looking, but it's the same behavior as other controllers: one write per tick of the loop.

This pattern guarantees that intermediate states will converge i.e. if the controller crashes between two writes, the behavior of the second write will be identical to if it hadn't crashed.

Comment (jveski) [reply internal/controllers/replication/symphony.go:54]: Added a comment

**Review by AYM1607 — APPROVED** [review]
LGTM

**Pull request merged**